### ISSUE TITLE
pyfunc_predict_fn parameter does not wio

### ISSUE THREAD

**Issue by manuelgilm** [mlflow/mlflow#11559]
### Issues Policy acknowledgement

- [X] I have read and agree to submit bug reports in accordance with the [issues policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md)

### Where did you encounter this bug?

Local machine

### Willingness to contribute

No. I cannot contribute a bug fix at this time.

### MLflow version

- mlflow, version 2.11.3

### System information

- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Microsoft Windows 11 Home
- **Python version**: Python 3.11.8
- **yarn version, if running the dev UI**:

### Describe the problem

I am able to log a sklearn model specifying the parameter `pyfunc_predict_fn`  as `predict_log_proba`. Then when I try to load the model back using the method `mlflow.pyfunc.load_model(model_uri)` I get an exception saying that  `MlflowException: Model implementation is missing required predict_log_proba method.` 

### Tracking information

```console
System information: Windows 10.0.22621
Python version: 3.11.8
MLflow version: 2.11.3
MLflow module location: [c:\Users\manue\projects\mlflow_for_ml_dev\.venv\Lib\site-packages\mlflow\__init__.py](file:///C:/Users/manue/projects/mlflow_for_ml_dev/.venv/Lib/site-packages/mlflow/__init__.py)
Tracking URI: file:///C:/Users/manue/projects/mlflow_for_ml_dev/mlruns
Registry URI: file:///C:/Users/manue/projects/mlflow_for_ml_dev/mlruns
Active experiment ID: 351285448386796167
Active run ID: 089593eb3d77480d9fa64e5f2ad07124
Active run artifact URI: file:///C:/Users/manue/projects/mlflow_for_ml_dev/mlruns/351285448386796167/089593eb3d77480d9fa64e5f2ad07124/artifacts
MLflow dependencies: 
  Flask: 3.0.2
  Jinja2: 3.1.3
  alembic: 1.13.1
  click: 8.1.7
  cloudpickle: 3.0.0
  docker: 7.0.0
  entrypoints: 0.4
  gitpython: 3.1.42
  graphene: 3.3
  importlib-metadata: 7.1.0
  markdown: 3.6
  matplotlib: 3.8.3
  numpy: 1.26.4
  packaging: 23.2
  pandas: 2.2.1
  protobuf: 4.25.3
  pyarrow: 15.0.2
  pytz: 2024.1
  pyyaml: 6.0.1
  querystring-parser: 1.2.4
  requests: 2.31.0
  scikit-learn: 1.4.1.post1
  scipy: 1.12.0
  sqlalchemy: 2.0.29
  sqlparse: 0.4.4
  waitress: 3.0.0
```

### Code to reproduce issue

```python
import mlflow 
from mlflow.models.signature import infer_signature
from sklearn.ensemble import RandomForestClassifier
from sklearn.datasets import load_iris 

# load the iris dataset
iris = load_iris(as_frame=True)
x = iris.data
y = iris.target

# infer signature
signature = infer_signature(model_input=x, model_output=y)
input_example = x.iloc[0:10]

experiment = mlflow.get_experiment_by_name("Default")

with mlflow.start_run(run_name="logging_with_custom_predict_function", experiment_id=experiment.experiment_id) as run:

    rfc = RandomForestClassifier()
    rfc.fit(x, y)

    # Log the model
    mlflow.sklearn.log_model(
        sk_model=rfc,
        artifact_path = rfc.__class__.__name__,
        signature=signature,
        input_example=input_example,
        pyfunc_predict_fn = "predict_log_proba"
        )
    
# load model using pyfunc
model_uri = f"runs:/{run.info.run_id}/{rfc.__class__.__name__}"
model = mlflow.pyfunc.load_model(model_uri)

# predict
model.predict(x)
```

### Stack trace

```
{
	"name": "MlflowException",
	"message": "Model implementation is missing required predict_log_proba method.",
	"stack": "---------------------------------------------------------------------------
MlflowException                           Traceback (most recent call last)
Cell In[1], line 33
     31 # load model using pyfunc
     32 model_uri = f\"runs:/{run.info.run_id}/{rfc.__class__.__name__}\"
---> 33 model = mlflow.pyfunc.load_model(model_uri)
     35 # predict
     36 model.predict(x)

File c:\\Users\\manue\\projects\\mlflow_for_ml_dev\\.venv\\Lib\\site-packages\\mlflow\\pyfunc\\__init__.py:873, in load_model(model_uri, suppress_warnings, dst_path, model_config)
    871     raise e
    872 predict_fn = conf.get(\"predict_fn\", \"predict\")
--> 873 return PyFuncModel(model_meta=model_meta, model_impl=model_impl, predict_fn=predict_fn)

File c:\\Users\\manue\\projects\\mlflow_for_ml_dev\\.venv\\Lib\\site-packages\\mlflow\\pyfunc\\__init__.py:627, in PyFuncModel.__init__(self, model_meta, model_impl, predict_fn)
    625 def __init__(self, model_meta: Model, model_impl: Any, predict_fn: str = \"predict\"):
    626     if not hasattr(model_impl, predict_fn):
--> 627         raise MlflowException(f\"Model implementation is missing required {predict_fn} method.\")
    628     if not model_meta:
    629         raise MlflowException(\"Model is missing metadata.\")

MlflowException: Model implementation is missing required predict_log_proba method."
}
```

### Other info / logs

```
REPLACE_ME
```

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [X] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

**Comment by B-Step62** [comment]
@manuelgilm Thank you so much for bringing this issue to our attention! 

This seems to be a regression introduced in MLflow 2.6.0. I've filed a patch PR and the issue should be fixed in the next release. For the meantime, you can invoke the `predict_log_proba` by directly accessing the underyling sklearn model

```
model = mlflow.sklearn.load_model(model_uri)
model.predict_log_proba(x)
```

**Comment by B-Step62** [comment]
Fixed by #11577
